### PR TITLE
Enable eslint rule to disallow awaiting a non-thenable type

### PR DESCRIPTION
**Pull request by robertbrignull**
Enables https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/await-thenable.md

I spotted that VS Code is giving suggestions for this error in the editor but not in CI. I guess it's not included in `@typescript-eslint/recommended`. Enabling this eslint rule means we can fix this case and avoid introducing more in the future.

I think this fix in `extension.ts` is a change of behaviour, because the [void operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/void) evaluates the expression and then throws away the result and returns `undefined`. So previously the `chosenAction === installActionName` condition was always false.

## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

**Review by aeisenberg — APPROVED** [review]

File: extensions/ql-vscode/src/extension.ts [hunk 1]
```diff
@@ -462,7 +462,7 @@ export async function activate(
     ) {
       registerErrorStubs([checkForUpdatesCommand], (command) => async () => {
         const installActionName = "Install CodeQL CLI";
-        const chosenAction = await void showAndLogErrorMessage(
+        const chosenAction = await showAndLogErrorMessage(
```
Comment (aeisenberg) [inline extensions/ql-vscode/src/extension.ts:465]: Interesting. I guess it was never possible to install the cli from this code path. I never tried this out, though so I don't really know.

Thanks for catching this.

**Comment by robertbrignull** [comment]
Oh and the CI found more cases which didn't show up locally for whatever reason. I'll look into them tomorrow.

**Comment by robertbrignull** [comment]
I've pushed more linting fixes. I believe all these extra changes are behaviour preserving.

**Comment by robertbrignull** [comment]
Ok, third time's the charm. Had to tix another alert that had been introduced on `main` since I opened this PR 🤦

**Review by koesie10 — APPROVED** [review]

**Pull request merged**